openedx/core/djangoapps/discussions/api/external_discussions.py

```diff
@@ -0,0 +1,93 @@
+import logging
+
+from opaque_keys.edx.keys import LearningContextKey, UsageKey
+
+from .data import ExternalDiscussionData
+from ..models import ExternalDiscussionsIdMapping
+
+log = logging.getLogger(__name__)
+
+
+def create_external_discussion_mapping(
+    context_key: LearningContextKey,
+    usage_key: UsageKey,
+    external_discussion_id: str
+) -> ExternalDiscussionData:
+    discussion_mapping = ExternalDiscussionsIdMapping.objects \
+        .create(
+            context_key=context_key,
+            usage_key=usage_key,
+            external_discussion_id=external_discussion_id
+        )
```

Comment:
Could you check for exceptions here? One might come up if violate uniqueness. That should be captured here and we should return a ValueError (I think that makes the most sense?)

Reply:
Catching `IntegrityError` and outputting a `ValueError`. Also added a test case for this